### ISSUE TITLE
CI: Fedora/CentOS based jobs are failing after Fedora 35 release

### ISSUE THREAD

**Issue by wainersm** [kata-containers/kata-containers#2960]
There are jobs which use the image-builder (`kata-containers/tools/osbuilder/image-builder`) to build the guest image. The image-builder is actually a container built out of Fedora. Currently it uses `fedora:latest`; Fedora 35 was released yesterday and that started to fail the build of the image-builder image (the fedora 35 dnf mirrors are broken).

Here is an error log:
```
00:58:07 + podman build --build-arg http_proxy= --build-arg https_proxy= -t image-builder-osbuilder /tmp/jenkins/workspace/kata-containers-2.0-centos-8-CRIO_K8S-main-baseline/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/image-builder
00:58:07 STEP 1: FROM registry.fedoraproject.org/fedora:latest
00:58:07 Trying to pull registry.fedoraproject.org/fedora:latest...
00:58:08 Getting image source signatures
00:58:08 Copying blob sha256:60d974a73230da76eb030c3439d8c859ada8f47a2a31e2a99516f24378edcb37
00:58:10 Copying config sha256:d08dfa41ca0b4449b2da67a85fee01d6d5901f1cb700edf6767a13d2de7dc252
00:58:10 Writing manifest to image destination
00:58:10 Storing signatures
00:58:12 STEP 2: RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
00:58:13 --> 9455ff83fc7
00:58:13 STEP 3: RUN dnf install -y qemu-img parted gdisk e2fsprogs gcc xfsprogs findutils
00:58:13 Fedora 35 - x86_64                              0.0  B/s |   0  B     00:00    
00:58:13 Errors during downloading metadata for repository 'fedora':
00:58:13   - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
00:58:13 Error: Failed to download metadata for repo 'fedora': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
00:58:14 Error: error building at STEP "RUN dnf install -y qemu-img parted gdisk e2fsprogs gcc xfsprogs findutils": error while running runtime: exit status 1
00:58:14 make: *** [Makefile:130: /tmp/jenkins/workspace/kata-containers-2.0-centos-8-CRIO_K8S-main-baseline/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/kata-containers.img] Error 125
00:58:14 [install_kata_image.sh:50] ERROR: sudo -E USE_DOCKER= DISTRO=ubuntu make -e image
00:58:14 [install_kata.sh:22] ERROR: .ci/install_kata_image.sh 
00:58:14 [jenkins_job_build.sh:172] ERROR: ci/setup.sh
```

**Comment by wainersm** [comment]
Cc @fgiudici @littlejawa @fidencio @marcel-apf @devimc @GabyCT

I'm coping you because that issue breaks the jobs you own. I've already a fix for that problem.

**Comment by wainersm** [comment]
@ariel-adam @GabyCT this issue is an opportunity to discuss further the "always green CI" initiative...

Some almost random thoughts:

* I only found that issue because I happened to open a PR, otherwise I would not have seen that. What would happen then?
* The job started to fail yesterday after the release of Fedora 35, our baseline jobs ran *before* that so I only see the error today. Maybe I should have ran the baseline job manually yesterday to check it was really fine?
* @jodh-intel thought the problem was network connective; I  bet he didn't look at the baseline jobs because we are not used to that (only after a while I came up with the great idea of checking the baseline jobs)
* I opened the issue on this repository because I had already a fix and known the fix should be applied on this repository it would be easier to reference on the commit message. However I could have opened on *tests* or *ci* repositories. So where should be issues affecting the baseline job opened?
*  On this issue I CC'ed the baseline owners being affected. How to ensure others will be doing the same
* I think we should be label for marking (and ease search) the issues related with baseline jobs

**Comment by jodh-intel** [comment]
We really shouldn't be using `fedora:latest` for https://github.com/kata-containers/kata-containers/blob/main/tools/osbuilder/image-builder/Dockerfile#L6 imho.

Our general practise is to pin versions to minimise breakages, as we've been broken _soooo_ many times by upstreams in the past. That's why we introduced the concept of "versions databases":

- https://github.com/kata-containers/kata-containers/blob/main/versions.yaml
- https://github.com/kata-containers/tests/blob/main/versions.yaml

However, osbuilder tends to encode the version number in one of the shell config files. The use of `latest` in the image builder may well be an oversight given we even pin the osbuilder Fedora rootfs version:

- https://github.com/kata-containers/kata-containers/blob/main/tools/osbuilder/rootfs-builder/fedora/config.sh#L8

**Comment by jodh-intel** [comment]
> I am wondering what else can we be doing to improve kata's code quality in general.... ideas?

That sounds like a topic for the AC meeting.

I don't think there is one simple answer to this, but timely code reviews (by as many people as possible), good docs and static checks are part of the solution.

If you're aware of a linter for Dockerfiles, please add it to

**Comment by jodh-intel** [comment]
Prolly worth creating a separate issue to discuss quality though.
